src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,505 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+            }));
         }
+    }, [mode, initialDiskParams]);
 
-        return validationFailed;
-    }
-
-    existingVolumeNameDelta(value, poolName) {
-        const { storagePools, vm } = this.state;
-        const stateDelta = { existingVolumeName: value };
-        const pool = storagePools.find(pool => pool.name === poolName && pool.connectionName === vm.connectionName);
-        if (!pool)
-            return stateDelta;
+    useEffect(() => {
+        // Follow up state updates after 'existingVolumeName' is changed
+        if (!diskParams.storagePool)
+            return;
 
-        stateDelta.format = getDefaultVolumeFormat(pool);
+        let format = getDefaultVolumeFormat(diskParams.storagePool);
         let deviceType = "disk";
-        if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
-            const volume = pool.volumes.find(vol => vol.name === value);
+        if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(diskParams.storagePool.type) > -1) {
+            const volume = diskParams.storagePool.volumes.find(vol => vol.name === diskParams.existingVolumeName);
             if (volume && volume.format) {
-                stateDelta.format = volume.format;
+                format = volume.format;
                 if (volume.format === "iso")
                     deviceType = "cdrom";
             }
         }
+        setDiskParams(diskParams => ({
+            ...diskParams,
+            format,
+            device: deviceType,
+        }));
+    }, [diskParams.storagePool, diskParams.existingVolumeName]);
 
-        this.onValueChanged("device", deviceType);
+    useEffect(() => {
+        // Initial state settings and follow up state updates after 'device' is changed
 
-        return stateDelta;
-    }
+        // According to https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms (section about 'target'),
+        // scsi is the default option for libvirt for cdrom devices
+        let newBus = diskParams.device === "cdrom" && "scsi";
+
+        if (!newBus) {
+            // If disk with the same device exists, use the same bus too
+            for (const disk of Object.values(vm.disks)) {
+                if (disk.device === diskParams.device) {
+                    newBus = disk.bus;
+                    break;
+                }
+            }
+            newBus = newBus || "virtio";
+        }
 
-    onValueChanged(key, value) {
-        let stateDelta = {};
-        const { storagePools, vm } = this.state;
+        setDiskParams(diskParams => ({ ...diskParams, busType: newBus }));
+    }, [diskParams.device, vm.disks]);
+
+    useEffect(() => {
+        // Initial state settings and follow up state updates after 'busType' is changed
+        const existingTargets = Object.getOwnPropertyNames(vm.disks);
+        const availableTarget = isMediaInsertion ? disk.target : getNextAvailableTarget(existingTargets, diskParams.busType);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231116-154931-f2ab7bbc-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)